packages/compile-solidity/compilerSupplier/loadingStrategies/LoadingStrategy.js

```diff
@@ -77,16 +77,27 @@ class LoadingStrategy {
     );
   }
 
+  markListeners() {
+    return {
+      uncaughtException: new Set(process.listeners("uncaughtException")),
+      unhandledRejection: new Set(process.listeners("unhandledRejection")),
+    };
+  }
+
   /**
-   * Cleans up error listeners set (by solc?) when requiring it. (This code inherited from
-   * previous implementation, note to self - ask Tim about this)
+   * Cleans up error listeners left by soljson
+   * Use with `markListeners()`
    */
-  removeListener() {
-    const listeners = process.listeners("uncaughtException");
-    const execeptionHandler = listeners[listeners.length - 1];
+  removeListener(markedListeners) {
+    for (const eventName of ["uncaughtException", "unhandledRejection"]) {
+      const listeners = process.listeners(eventName);
+      const newListeners = listeners.filter(
+        (listener) => !markedListeners[eventName].has(listener),
+      );
 
-    if (execeptionHandler) {
-      process.removeListener("uncaughtException", execeptionHandler);
+      for (const listener of newListeners) {
+        process.removeListener(eventName, listener);
+      }
```

Comment:
This could be condensed to:

```javascript
for (const eventName in markedListeners) {
  const marked= markedListeners[eventName];
  process.listeners(eventName).forEach((listener) => {
    if (!marked.has(listener)) {
      process.removeListener(eventName, listener);
    }
  });
}
```

And if we change `markedListeners` to a `Map<string, Array<AnyFunction>` it could be:

```javascript
for (const [eventName, marked] of markedListeners) {
  process.listeners(eventName).forEach((listener) => {
    if (!marked.has(listener)) {
      process.removeListener(eventName, listener);
    }
  });
}
```

Reply:
Borrowed the first one :) Thanks @davidmurdoch!